src/jemalloc.c

```diff
@@ -2384,6 +2390,8 @@ je_malloc(size_t size) {
 	 * 0) in a single branch.
 	 */
 	uint64_t allocated_after = allocated + usize;
+	assert((!malloc_initialized() && threshold == 0) ||
+	    (ind == sz_size2index(size) && usize == sz_index2size(ind)));
```

Comment:
Is it the case that the part `(ind == sz_size2index(size) && usize == sz_index2size(ind))` always holds, regardless of `malloc_initialized()`?

I think we can also consolidate the assertion here with the one above, into a big (verbose) one here, like:
```
assert(ind == sz_size2index(size) && usize == sz_index2size(ind));
if (malloc_initialized()) {
    assert(usize > 0);
} else {
    assert(ind == 0 && usize == 0 && threshold == 0);
}
```

Reply:
I should better document this; we have to be conservative, as there may be concurrent threads doing malloc_init(), i.e. even if !initialized(), we may still observe non zero ind / usize (as in, another thread is doing malloc_init and writing to the lookup tables).

Let me polish and add comments.

Reply:
The other thing is, we cannot really call `sz_size2index` / `sz_index2size` if not initialized, they have assertions that would fail.